examples/simple_key_value_store.rs

```diff
@@ -139,294 +144,156 @@ struct CliArgs {
-                        info!("ClientManager of {:?} forwarding data to DataManager around {:?}",
-                            client_name, plain_data.name());
-                        return Ok(vec![MethodCall::Put {
-                            destination: plain_data.name(),
-                            content: Data::PlainData(plain_data) }]);
-                    },
-                    Authority::NaeManager(group_name) => {
-                        info!("DataManager of {:?} asked to put plain data named {:?}",
-                            group_name, plain_data.name());
-                        assert_eq!(group_name, plain_data.name());
-                        let data_name = plain_data.name();
-                        let _ = self.db.entry(plain_data.name())
-                            .or_insert(plain_data);
-                        assert!(self.db.contains_key(&data_name));
-                    },
-                    _ => {}
-                }
-            },
-            _ => {}
-        };
-        return Ok(vec![]);
+    else if cmds.len() == 1 && cmds[0] == "exit" {
+        return Some(UserCommand::Exit);
     }
-
-    fn handle_post(&mut self, _our_authority: Authority, _from_authority: Authority,
-        _from_address: types::SourceAddress, _dest_address: types::DestinationAddress,
-        _data: Data) -> Result<Vec<MethodCall>, InterfaceError> {
-        Err(InterfaceError::Abort)
+    else if cmds.len() == 2 && cmds[0] == "get" {
+        return Some(UserCommand::Get(cmds[1].to_string()));
     }
-
-    fn handle_refresh(&mut self, _type_tag: u64, _from_group: NameType, _payloads: Vec<Vec<u8>>) {
-        unimplemented!()
+    else if cmds.len() == 3 && cmds[0] == "put" {
+        return Some(UserCommand::Put(cmds[1].to_string(), cmds[2].to_string()));
     }
 
-    fn handle_get_response(&mut self, from_address: NameType, response: Data) -> Vec<MethodCall> {
-        println!("testing node received get_response from {} with data {:?}", from_address, response);
-        vec![]
-    }
+    None
+}
 
-    fn handle_put_response(&mut self, _from_authority : Authority, _from_address: types::SourceAddress,
-                                      response: ResponseError) -> Vec<MethodCall> {
-        println!("testing node received error put_response {}", response);
-        vec![]
-    }
+////////////////////////////////////////////////////////////////////////////////
+struct Client {
+    routing          : Routing,
+    routing_receiver : Receiver<Event>,
+    user_receiver    : Receiver<UserCommand>,
+    is_done          : bool,
+}
 
-    fn handle_post_response(&mut self, _from_authority: Authority,
-        _from_address: types::SourceAddress, _response: ResponseError)
-        -> Vec<MethodCall> {
-        unimplemented!();
-    }
+impl Client {
+    fn new(_bootstrap_peers: Vec<Endpoint>) -> Result<Client, RoutingError> {
+        let (routing_sender, routing_receiver) = mpsc::channel::<Event>();
```

Comment:
just a style issue to consider: naming a channel WHAT_sender, and WHAT_receiver is more clear than WHO_sender/_receiver, as it feels to me less clear the direction of the channel